docker/Dockerfile

```diff
@@ -83,15 +87,17 @@ RUN set -e; \
     esac; \
     case "$DISTRO_ID-$DISTRO_VERSION_ID" in \
     'debian-'* | 'linuxmint-'* | 'ubuntu-'*) \
-        curl --retry "$DEB_MAX_ATTEMPT" --retry-connrefused --retry-delay 3 -fsSL 'https://bazel.build/bazel-release.pub.gpg' \
-        | gpg --dearmor \
-        | sudo tee /etc/apt/trusted.gpg.d/bazel.gpg \
-        >/dev/null; \
-        echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" \
-        | sudo tee /etc/apt/sources.list.d/bazel.list; \
-        sudo apt-get autoremove -y; \
-        sudo apt-get clean; \
-        sudo rm -rf /var/lib/apt/lists/*; \
+        if [ "$TARGETARCH" == 'amd64' ]; then \
```

Comment:
Add a _ prefix on both side, I've seen weird cases where empty string in comparison doesn't work

Reply:
We require TARGETARCH to be non-empty. This is auto set if using Docker BuildKit

Reply:
No it won't, you're relying on an assumption in mind that no one else know, and nothing stops other code to implicitly change this env var.

Reply:
Then it will make the build wrong. Of course we can set whatever ARGS you like, but resetting an automatic ARG is not ideal. Especially when they are meant for this purpose.

https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope

If it is empty, then we don't know what the target architecture is. We may default it to amd64, but a better solution is just using BuildKit. (or docker buildx)

Reply:
Yes your opinion is right, but you don't have control on this, you are describing a best practice that you can't really enforce automatically, and that's where surprises come.
What I proposed here is to provide a clear failure point if something goes wrong, having a prefix can ensure anything goes wrong, your error is on comparison, not on something strange.

Reply:
> We may default it to amd64, but a better solution is just using BuildKit.

Buildkit has a lot of unique features, a lot of bugs, and a lot of breaking changes. And there're still many workflow doesn't work with buildkit as well. So don't put a hard dependency on it for nothing. The default value feature you want can be easily done in shell, but for buildkit even if it works it's probably an UB given there's barely any doc available.